transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/insight/SelectTransferMethodInsightTest.java

```diff
@@ -0,0 +1,159 @@
+        IdlingRegistry.getInstance().register(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @After
+    public void cleanup() {
+        TransferMethodRepositoryFactory.clearInstance();
+        UserRepositoryFactory.clearInstance();
+        IdlingRegistry.getInstance().unregister(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @Test
+    public void testSelectTransferMethod_verifyInsightEventCreatedOnTransferMethodOptionsLoad() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("user_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_keys_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                timeout(5000).times(1)).trackImpression(any(Context.class),
+                eq(HyperwalletInsight.PAGE_TRANSFER_METHOD_SELECT),
+                eq(HyperwalletInsight.TRANSFER_METHOD_GROUP),
+                mParamsCaptor.capture());
+
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_country"), is("US"));
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_currency"), is("USD"));
+    }
+
+    @Test
+    public void testSelectTransferMethod_verifyInsightEventCreatedOnCountrySelection() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("user_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_keys_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                times(1)).trackImpression(any(Context.class),
+                eq(HyperwalletInsight.PAGE_TRANSFER_METHOD_SELECT),
+                eq(HyperwalletInsight.TRANSFER_METHOD_GROUP),
+                mParamsCaptor.capture());
+
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_country"), is("US"));
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_currency"), is("USD"));
+
+        onView(withId(R.id.select_transfer_method_country_value)).perform(click());
+        onView(allOf(withId(R.id.country_name), withText("Canada"))).perform(click());
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                timeout(5000).times(2)).trackImpression(any(Context.class),
+                eq(HyperwalletInsight.PAGE_TRANSFER_METHOD_SELECT),
+                eq(HyperwalletInsight.TRANSFER_METHOD_GROUP),
+                mParamsCaptor.capture());
+
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_country"), is("CA"));
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_currency"), is("CAD"));
```

Comment:
I think  here,  we can also check the country selection trackclick() event.

Reply:
We will do this in another PR, thanks.